### PR TITLE
tktable: fix remaining compiler warnings

### PR DESCRIPTION
**Pull request by chrstphrchvz**
#### Description
Tktable has been dormant for several years, so for now I'd like to incorporate these patches into MacPorts to have one place they're publicly posted and in use. I will likely incorporate some of these changes into the Perl Tk::TableMatrix module, which I am working on submitting to MacPorts.

The upcoming Tcl/Tk 8.6.11 release contains a fix for Tktable running on Tk Aqua, but requires another patch in Tktable to take advantage of; I will try submitting the patch when Tcl/Tk 8.6.11 is released.

I will consider seeing if Tcl/Tk developers are interested publishing a maintenance release of Tktable.

###### Tested on

macOS 10.15.7
Xcode 12.1 command line tools

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: x11/tktable/files/patch-INT2PTR.diff [hunk 1]
```diff
@@ -0,0 +1,798 @@
...
+ 
+ 
+ #--------------------------------------------------------------------
+ # Everyone should be linking against the Tcl stub library.  If you
+ # can't for some reason, remove this definition.  If you aren't using
+
+Index: configure.in
+==================================================================
+--- configure.in
++++ configure.in
```
Comment (ryandesign) [inline x11/tktable/files/patch-INT2PTR.diff:345]: A patch that modifies both configure and configure.in, in that order, can cause autoconf to be run unexpectedly (and that could fail) if the clock second when configure gets modified is not the same as the clock second when configure.in gets modified. The solution is to either patch only configure, or patch only configure.in and run autoreconf, or patch configure.in before patching configure (list configure.in before configure in the patchfile).

Comment (chrstphrchvz) [reply x11/tktable/files/patch-INT2PTR.diff]: Thank you for pointing this out. I have opted for patching configure before configure.in.

Comment (chrstphrchvz) [reply x11/tktable/files/patch-INT2PTR.diff]: configure.in before configure, sorry

**Comment by chrstphrchvz** [comment]
Temporarily marking as draft to incorporate another change needed on some platforms (e.g. Linux aarch64)

**Comment by chrstphrchvz** [comment]
Latest patch failure due to conflict between patch-INT2PTR.diff and patch-quartz.diff has been addressed.

**Review by pes2020kk — COMMENTED** [review]
ios 14

**Review by pes2020kk — COMMENTED** [review]
hi

**Pull request merged**